src/core/pointcloud/qgspointcloudstatistics.cpp

```diff
@@ -111,3 +127,54 @@ void QgsPointCloudStatistics::combineWith( const QgsPointCloudStatistics &stats
   }
   mSampledPointsCount += stats.mSampledPointsCount;
 }
+
+QJsonObject QgsPointCloudStatistics::toJson() const
+{
+  QJsonObject obj;
+  obj.insert( QStringLiteral( "sampled-points" ), QJsonValue::fromVariant( sampledPointsCount() ) );
+  QJsonObject stats;
+  for ( QString &attr : mStatisticsMap.keys() )
+  {
+    QgsPointCloudStatistics::AttributeStatistics stat = mStatisticsMap.value( attr );
+    stats.insert( attr, QgsPointCloudStatistics::fromStatsToJson( stat ) );
+  }
+  obj.insert( QStringLiteral( "stats" ), stats );
+  return obj;
+}
+
+QJsonObject QgsPointCloudStatistics::fromStatsToJson( QgsPointCloudStatistics::AttributeStatistics &stats )
+{
+  QJsonObject obj;
+  obj.insert( QStringLiteral( "minimum" ), stats.minimum );
+  obj.insert( QStringLiteral( "maximum" ), stats.maximum );
+  obj.insert( QStringLiteral( "mean" ), stats.mean );
+  if ( !std::isnan( stats.stDev ) )
+  {
+    obj.insert( QStringLiteral( "stDev" ), stats.stDev );
```

Comment:
it would be good to unify naming convention - kebab case ("sampled-points", "class-count") or camel case ("stDev"). My personal preference is the former...